crates/bevy_ecs/src/query/state.rs

```diff
@@ -1196,7 +1304,10 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
     #[track_caller]
     #[inline]
     pub fn single<'w>(&mut self, world: &'w World) -> ROQueryItem<'w, Q> {
-        self.get_single(world).unwrap()
+        match self.get_single(world) {
+            Ok(items) => items,
+            Err(error) => panic!("Cannot get single mutable query result: {error}"),
+        }
```

Comment:
```rs
        self.get_single(world).expect(format!("Cannot get single mutable query result: {error}"))
```

I think you can achieve the same by using the `expect()` method, so you can avoid such verbose match block for simply panicking on error. Same thing for the rest of the files.

I used format macro here but I think you could also use string concatenation or whatever works best to display the full error.

Reply:
That would be less performant, since it would do the string formatting regardless of whether an error occurred.

Reply:
I don't believe that's the case, because I'm trying to extract both the `Err(error)` and `Ok(item)` variants. To my knowledge, `.expect(...)` doesn't use the `Display` implementation for the error message with a custom format. If you check this [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2b34b7f9bd813fb3eeb1a12d9a4306f3), you can see the display implementation isn't used, which was specifically requested behaviour from an earlier review.

Reply:
Disregard then! I wasn't considering those very valid points and I probably missed that earlier review too.